File: Misc/130_lc_1451.java
import java.util.Arrays;

/*
 * 1451. Rearrange Words in a Sentence
 */

class Solution {
    public String arrangeWords(String text) {
        // Convert to lowercase
        text = text.toLowerCase();
        String[] words = text.split(" ");
        Arrays.sort(words, (a,b) -> Integer.compare(a.length(), b.length()));
        StringBuilder res = new StringBuilder(String.join(" ", words));

        res.setCharAt(0, Character.toUpperCase(res.charAt(0)));
        
        return res.toString();
    }
}

File: Misc/130_lc_1451.py
'''
1451. Rearrange Words in a Sentence
'''

class Solution:
    def arrangeWords(self, text: str) -> str:
        text = text.lower()
        text = list(text.split(" "))
        text.sort(key =lambda x : len(x))
        res = " ".join(text)
        res = res[0].upper() + res[1:]

        return res